### PR TITLE
Issue 6 upgrade eslint

### LINKED ISSUE #67: Update deprecated react life-cycle functions.
# 🐛 Description

At least `ColonyEditorContainer` and `ColonyEditor` use the deprecated `componentWillReceiveProps` method.

All deprecated life-cycle methods should be updated.

# 🔢 To reproduce

1. `npm run test`
2. See warnings flash by

# ✌🏽 Expected behaviour

- [ ] we don't use `componentWillReceiveProps` or any other deprecated functions

# 💻 Traceback

```console
WARN: 'Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: %s', 'ColonyEditorContainer' 'ColonyEditor'
```

### PR DESCRIPTION
**Pull request by skymandr**
This upgrades `eslint` to the 7.x version (latest is 8.x, but that isn't compatible yet), which requires upgrading `eslint-config-aribnb`, which unfortunately unlocks a whole hell of lint errors. I have fixed many of them, but notably I have ignored `react/jsx-props-no-spreading` and `react/destructuring-assignment`, with more than 200 instances each. See #133 for details.

Part of #6 and resolves #67.

### CODE REVIEW

File: scanomatic/ui_server_data/js/ccc/components/ColonyEditor.jsx [hunk 1]
```diff
@@ -17,7 +17,8 @@ export default class ColonyEditor extends React.Component {
     this.handleUpdate = this.handleUpdate.bind(this);
   }
 
-  componentWillReceiveProps(props) {
+  // eslint-disable-next-line camelcase
+  UNSAFE_componentWillReceiveProps(props) {
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/components/ColonyEditor.jsx:21]: Does this mean you solved the #67

File: scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx [hunk 2]
```diff
@@ -25,8 +25,9 @@ export default class CCCEditorContainer extends React.Component {
   }
 
   handleFinishUpload(newImage) {
-    const plates = [...this.state.plates];
-    for (let i = 1; i <= this.state.platesPerImage; i += 1) {
+    const prevState = this.state;
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx:28]: it should be considered the current state here and not previous state

**Review by local-minimum — CHANGES_REQUESTED** [review]

File: scanomatic/ui_server_data/js/ccc/components/ColonyEditor.jsx (hunk 1 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/components/ColonyEditor.jsx:21]: Looks like it, yeah.

File: scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx (hunk 2 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx:28]: Changed the name, but only in this instance. Let me know if you think it is badly named elsewhere as well.

File: scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx [hunk 3]
```diff
@@ -26,30 +25,34 @@ export default class CCCEditorContainer extends React.Component {
...
       currentPlate,
     });
   }
 
   handleFinishPlate() {
-    const currentPlate =
-            this.state.currentPlate < this.state.plates.length - 1
-              ? this.state.currentPlate + 1
-              : null;
+    const prevState = this.state;
```
[the remark this answers is not in the document]
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/containers/CCCEditorContainer.jsx]: fixed

File: scanomatic/ui_server_data/js/ccc/containers/ColonyEditorContainer.jsx [hunk 4]
```diff
@@ -58,7 +49,9 @@ export default class ColonyEditorContainer extends React.Component {
   }
 
   handleUpdate(data) {
-    const colonyData = Object.assign({}, this.state.colonyData, data);
+    const prevState = this.state;
```
[the remark this answers is not in the document]
Comment (skymandr) [reply scanomatic/ui_server_data/js/ccc/containers/ColonyEditorContainer.jsx]: fixed

**Review by local-minimum — APPROVED** [review]

**Pull request merged**